src/ring_theory/integral_domain.lean

```diff
@@ -131,6 +131,37 @@ end
 
 end
 
+section euclidean_division
+
+namespace polynomial
+
+open_locale polynomial
+
+variables (K : Type) [field K] [algebra R[X] K] [is_fraction_ring R[X] K]
+
+lemma exists_quotient_remainder (f : R[X]) {g : R[X]} (hg : g.monic) :
+  ∃ q r : R[X], r.degree < g.degree ∧ f = q * g + r :=
+begin
+  refine ⟨f /ₘ g, f %ₘ g, _, _⟩,
+  { exact degree_mod_by_monic_lt _ hg },
+  { rw [add_comm, mul_comm, mod_by_monic_add_div f hg] },
+end
+
+lemma div_eq_quo_add_rem_div (f : R[X]) {g : R[X]} (hg : g.monic) :
+  ∃ q r : R[X], r.degree < g.degree ∧ (↑f : K) / ↑g = ↑q + ↑r / ↑g :=
+begin
+  refine ⟨f /ₘ g, f %ₘ g, _, _⟩,
+  { exact degree_mod_by_monic_lt _ hg, },
+  { have hg' : (↑g : K) ≠ 0 := by exact_mod_cast (monic.ne_zero hg),
+    field_simp [hg'],
+    norm_cast,
+    rw [add_comm, mul_comm, mod_by_monic_add_div _ hg], },
+end
```

Comment:
I was thinking along the lines of the following prove, where we use `exists_quotient_remainder` with a few almost automatic rewrites. It's not really shorter in number of lines but definitely shorter in the amounts of ideas.
```suggestion
begin
  obtain ⟨q, r, hr, rfl⟩ := exists_quotient_remainder f hg,
  refine ⟨q, r, hr, _⟩,
  have hg' : (↑g : K) ≠ 0 := by exact_mod_cast (monic.ne_zero hg),
  field_simp [hg'],
  norm_cast
end
```

Reply:
That makes sense, and probably helps with readability as well (it's a bit clearer what's going on). I'm still not fully clear as to why we need this result though...